libraries/AP_Scheduler/AP_Scheduler.cpp

```diff
@@ -114,6 +121,12 @@ void AP_Scheduler::init(const AP_Scheduler::Task *tasks, uint8_t num_tasks, uint
     perf_info.set_loop_rate(get_loop_rate_hz());
     perf_info.reset();
 
+    if (_options & uint8_t(Options::RECORD_TASK_INFO)) {
+        perf_info.allocate_task_info(_num_tasks);
+    } else {
+        perf_info.free_task_info();
```

Comment:
Uh - how many times were you expecting `init()` to be called?

Reply:
Yeah this was @tridge 's idea. @tridge - to support this dynamically I would have to add an update function that checks for the changed parameter - I think it would be better to simply require a reboot.

Reply:
Ok we actually have an update_logging function so I have piggy-backed on that